crates/pagecache/src/util.rs

```diff
@@ -57,5 +57,5 @@ pub(crate) fn maybe_decompress(buf: Vec<u8>) -> std::io::Result<Vec<u8>> {
     }
 
     #[cfg(not(feature = "compression"))]
-    Ok(buf)
+    Ok(buf.to_vec())
```

Comment:
This forces all data going through this path to be copied when compression is disabled. Please keep this argument as requiring an owned `Vec` so this copy is avoided when compression is disabled.

Reply:
Oh yes, sure! I wasn't aware of this as I am relatively new to Rust. Thanks for the hint :+1:

Reply:
In your usage of it, it's actually fine because it only gets called if compression is enabled anyway (and clippy was right to point it out as a possible optimization), but this makes it a little more defensive in case something changes over time.

The main thing to be careful of in a database is anything that takes a buffer and makes a duplicate of it. Avoiding this is the single biggest performance trick in database engineering lol